www/js/diary/infinite_scroll_list.js

```diff
@@ -319,45 +377,95 @@ angular.module('emission.main.diary.infscrolllist',['ui-leaflet',
       $scope.allTrips = true;
     }
 
-    $scope.recomputeDisplayTimelineEntries();
+    $scope.recomputeListEntries();
     // scroll to the bottom while changing filters so users don't have to
     // fixes the first of the fit-and-finish issues from
     // https://github.com/e-mission/e-mission-docs/issues/662
     $ionicScrollDelegate.scrollBottom();
     ClientStats.addReading(ClientStats.getStatKeys().LABEL_TAB_SWITCH, {"source": prev, "dest": $scope.getActiveFilters()});
   }
 
-  $scope.recomputeDisplayTimelineEntries = function() {
+  $scope.updateLoadedRange = (oldestTrip, latestTrip) => {
+    if (oldestTrip && oldestTrip.start_ts <= $scope.infScrollControl.pipelineRange.start_ts) {
+      Logger.log("Oldest trip in batch starts at "+ moment(oldestTrip.start_ts)
+          +" pipeline starts at "+moment($scope.infScrollControl.pipelineRange.start_ts)
+          +". Reached the start of pipeline, nothing earlier to load");
+      $scope.infScrollControl.oldestLoadedTs = oldestTrip.start_ts;
+      $scope.infScrollControl.reachedPipelineStart = true;
+    }
+    if (latestTrip && latestTrip.end_ts >= $scope.infScrollControl.pipelineRange.end_ts) {
+      Logger.log("Latest trip in batch ends at "+ moment(latestTrip.end_ts)
+          +" pipeline ends at "+moment($scope.infScrollControl.pipelineRange.end_ts)
+          +". Reached the end of pipeline, nothing later to load");
+      $scope.infScrollControl.reachedPipelineEnd = true;
+      $scope.infScrollControl.latestLoadedTs = latestTrip.end_ts;
+    }
+  }
+
+  $scope.updateTimeRange = (dir, oldestTrip, latestTrip) => {
+    $scope.updateLoadedRange(oldestTrip, latestTrip);
+    if (!dir || dir == 'past') {
+      if (oldestTrip) {
+          if (oldestTrip.start_ts > $scope.infScrollControl.pipelineRange.start_ts) {
```

Comment:
isn't this the same as the `reachedPipelineStart` check?

Reply:
TODO - maybe remove this redundant check

Reply:
Cleanup: this is no longer relevant in the latest branch of the migration, as `updateTimeRange()` isn't needed anymore